library/alloc/src/rc.rs

```diff
@@ -2512,14 +2512,23 @@ trait RcInnerPtr {
     fn inc_strong(&self) {
         let strong = self.strong();
 
+        // We insert an `assume` here to hint LLVM at an otherwise
+        // missed optimization.
+        // SAFETY: The reference count will never be zero when this is
+        // called.
+        unsafe {
+            core::intrinsics::assume(strong != 0);
+        }
```

Comment:
```suggestion
```
According to https://rust.godbolt.org/z/Wbn6jMcTb `core::intrinsics::assume(strong != 0)` is completely unnecessary as LLVM is able to optimize as well without.

Reply:
Yeah, this doesn't have any impact on this particular function. I've included it because the original code had a check for it, which I assume was meant to help LLVM optimize surrounding code when inlined. Based on PR #53080 I believe one example of this is optimizing generated drop code in some situations.